### PR TITLE
[SYCL][MLIR] Add stream type.

### PR DESCRIPTION
**Pull request by arnamoy10**
`clang` generates the following for stream type

`%"class.sycl::_V1::stream" = type { [16 x i8], %"class.sycl::_V1::accessor", %"class.sycl::_V1::accessor.1", %"class.sycl::_V1::accessor", i32, i64, i32, i32, i32, i32 }
%"class.sycl::_V1::accessor.1" = type { %"class.sycl::_V1::detail::AccessorImplDevice", %union.anon.4 }
%"class.sycl::_V1::detail::AccessorImplDevice" = type { %"class.sycl::_V1::id", %"class.sycl::_V1::range", %"class.sycl::_V1::range" }
%"class.sycl::_V1::id" = type { %"class.sycl::_V1::detail::array" }
%"class.sycl::_V1::detail::array" = type { [1 x i64] }
%"class.sycl::_V1::range" = type { %"class.sycl::_V1::detail::array" }
%union.anon.4 = type { i32 addrspace(1)* }
%"class.sycl::_V1::accessor" = type { %"class.sycl::_V1::detail::AccessorImplDevice", %union.anon }
%union.anon = type { i8 addrspace(1)* }`

### CODE REVIEW

File: mlir-sycl/test/Conversion/SYCLToLLVM/sycl-types-to-llvm.mlir [hunk 1]
```diff
@@ -241,6 +241,20 @@ func.func @test_sub_group(%arg0: !sycl.sub_group) {
...
..., (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i8, 1>)>)>, struct<"class.sycl::_V1::accessor.1.1", (struct<"class.sycl::_V1::detail::AccessorImplDevice.1", (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i32, 1>)>)>, struct<"class.sycl::_V1::accessor.1", (struct<"class.sycl::_V1::detail::AccessorImplDevice.1", (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i8, 1>)>)>, i32, i64, i32, i32, i32, i32)>) {
```
Comment (victor-eds) [inline mlir-sycl/test/Conversion/SYCLToLLVM/sycl-types-to-llvm.mlir:251]:
```suggestion
// CHECK: llvm.func @test_stream(%arg0: !llvm.struct<"class.sycl::_V1::stream", (array<16 x i8>, struct<"class.sycl::_V1::accessor.1", (struct<"class.sycl::_V1::detail::AccessorImplDevice.1", (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i8, 1>)>)>, struct<"class.sycl::_V1::accessor.1.1", (struct<"class.sycl::_V1::detail::AccessorImplDevice.1", (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i32, 1>)>)>, struct<"class.sycl::_V1::accessor.1", (struct<"class.sycl::_V1::detail::AccessorImplDevice.1", (struct<"class.sycl::_V1::id.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>, struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>)>, struct<(ptr<i8, 1>)>)>, i32, i64, i32, i32, i32, i32)>) {
```
NIT

**Review by victor-eds — COMMENTED** [review]
Just a NIT

**Review by victor-eds — APPROVED** [review]

File: mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp [hunk 2]
```diff
@@ -182,6 +182,10 @@ SYCLOpAsmInterface::getAlias(mlir::Type Type, llvm::raw_ostream &OS) const {
                << Ty.getDataType() << "_";
             return AliasResult::FinalAlias;
           })
+      .Case<mlir::sycl::StreamType>([&](auto Ty) {
+        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_";
+        return AliasResult::OverridableAlias;
```
Comment (whitneywhtsang) [inline mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp:187]: Why is this `OverridableAlias`? Is there another way to write `sycl::stream`?

Comment (victor-eds) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp:187]: Missed that. I don't think so. Can address in a different PR.

Comment (whitneywhtsang) [reply mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp:187]: @arnamoy10 please add it in 
```
      .Case<mlir::sycl::AssertHappenedType, mlir::sycl::BFloat16Type,
            mlir::sycl::KernelHandlerType>([&](auto Ty) {
        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_";
        return AliasResult::FinalAlias;
      })
```

**Pull request merged**